src/handler.ts

```diff
@@ -9,8 +10,14 @@ import {
   publishEmployeesCelebrations,
   pusblishBankHolidays,
 } from './publisher';
+import { getSecret, initializeSecretsManager } from './secrets';
 
 export async function main() {
+  initializeSecretsManager();
+
+  const value = await getSecret('BAMBOOHR_KEY');
```

Comment:
move this out of the handler. This way this will be executed everytime the lambda is called (not only on lambda container startup) so if you have a lot of request it will be more expensive.

Reply:
@w3dani , where should I place that code? I mean, where is the right place to setup the code that only needs to be called on the lambda startup?

Reply:
I can't do that, since `getSecret()` is an async function. Is there a workaround for that? 🤔

Reply:
Just an idea, but you could do something like this:

```
const secrets;
initializeSecretsManager();

const getSecretsFromAWS = async () => {
 if(secrets) return;
 secrets = await getSecrets()
}

export async function main() {
  await getSecretsFromAWS();
  sdk.auth(value ?? '');
  ...
}
```

Reply:
I've somehow implemented what you suggest. Let me know what you think, @w3dani